packages/common/src/hardhat/tasks/compile.js

```diff
@@ -39,10 +39,7 @@ internalTask(TASK_COMPILE_SOLIDITY_GET_SOURCE_PATHS, async (_, { config, network
 
   return [
     ...filePaths,
-    `${ovmPath}/insured-bridge/implementation/L1_BridgePool.sol`,
```

Comment:
@chrismaree I removed this file from this array because (1) it uses solidity 0.8 in order to import the @openzeppelin/ERC20 contract and (2) we presumably won't need to compile+deploy it to the OVM. Is this the right way to do this?

Reply:
If this file Is still within the `contracts-ovm` directory, then this is not the right way to do this. What this modification is doing is appending additional contracts to the compilation step for hardhat. what you've done by removing this line is to actually ignore this contract from compilation in totality. To remove it from the `optimism` compilation, you simply need to remove this contract name from the hardhat config. However, the _only_ file currently in the hardhat config is `L2_BridgeDepositBox.sol` so this is the only contract that will be OVM compiled.

What this code is doing is basically enabling us to add contracts that hardhat sees to both the default `contracts` directory, as well as the `contracts-ovm` directory. I will add a follow-up PR soon to remove this logic where we have to specify each import line by line and will rather just pull in _all_ files within the `contracts-ovm` directory.

Reply:
i.e you should _not_ remove this file from here and leave this `compilation.js` task as it is to ensure this file gets compiled correctly. I've re-worked how this kind of selective compilation works in another PR https://github.com/UMAprotocol/protocol/pull/3255. once that is in this will be a lot cleaner. for now what you should do is just revert the changes on this file and compile these contracts in hardhat.

Reply:
Reverted